### PR TITLE
Add new CronJobRun model; add instances when cron jobs are executed

### PR DESCRIPTION
**Pull request by acrellin**

### CODE REVIEW

**Comment by acrellin** [comment]
@stefanv this is the PR I was referring to -- whenever you have a moment, I'd love a review

File: app/models.py [hunk 1]
```diff
@@ -1157,3 +1157,17 @@ def is_readable_by(self, user_or_token):
...
+
+
+class CronJobRun(Base):
+    """A record of a run (or attempted run) of a cron job."""
+    script = sa.Column(
+        sa.String,
+        nullable=False,
+        doc="Name of script being run.",
+    )
+    exit_status = sa.Column(
```
Comment (stefanv) [inline app/models.py:1169]: Should we also log stdout/stderr?

Comment (acrellin) [reply app/models.py:1169]: @stefanv  Good call - just pushed up that change

File: services/cron/cron.py [hunk 2]
```diff
@@ -78,10 +80,18 @@ def cache_to_file(self):
...
             tc.reset(key)
             try:
-                subprocess.Popen(
+                proc = subprocess.Popen(
                     script, stdout=subprocess.PIPE, stderr=subprocess.STDOUT
                 )
+                output, _ = proc.communicate()
             except Exception as e:
                 log(f'Error executing {script}: {e}')
+                DBSession().add(CronJobRun(script=script, exit_status=1))
```
Comment (stefanv) [inline services/cron/cron.py:89]: Add the exception string here as the proc_output?

**Review by stefanv — APPROVED** [review]
I suggest adding the exception string, if we don't get output.

Go ahead and merge if you're happy with that change and have made it.

File: app/models.py (hunk 1 above)
Comment (acrellin) [reply app/models.py:1169]: I tested that it captures both stdout and stderr

File: services/cron/cron.py (hunk 2 above)
Comment (acrellin) [reply services/cron/cron.py]: Done - renamed proc_output to output, and just store the exception string there in that case

**Comment by acrellin** [comment]
Thanks, @stefanv !

**Pull request merged**